Entities/SLBackground.cpp

```diff
@@ -0,0 +1,255 @@
+#include "SLBackground.h"
+#include "FrameMan.h"
+#include "SceneMan.h"
+#include "SettingsMan.h"
+
+namespace RTE {
+
+	ConcreteClassInfo(SLBackground, SceneLayer, 0);
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void SLBackground::Clear() {
+		m_Bitmaps.clear();
+		m_FrameCount = 1;
+		m_Frame = 0;
+		m_SpriteAnimMode = SpriteAnimMode::NOANIM;
+		m_SpriteAnimDuration = 1000;
+		m_SpriteAnimIsReversingFrames = false;
+		m_SpriteAnimTimer.Reset();
+		m_IsAnimatedManually = false;
+		m_AutoScrollX = false;
+		m_AutoScrollY = false;
+		m_AutoScrollStep.Reset();
+		m_AutoScrollStepInterval = 0;
+		m_AutoScrollStepTimer.Reset();
+		m_AutoScrollOffset.Reset();
+		m_FillColorLeft = ColorKeys::g_MaskColor;
+		m_FillColorRight = ColorKeys::g_MaskColor;
+		m_FillColorUp = ColorKeys::g_MaskColor;
+		m_FillColorDown = ColorKeys::g_MaskColor;
+
+		m_LayerScaleFactors = { Vector(1.0F, 1.0F), Vector(1.0F, 1.0F), Vector(2.0F, 2.0F) };
+		m_IgnoreAutoScale = false;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int SLBackground::Create() {
+		SceneLayer::Create();
+
+		m_Bitmaps.clear();
+		m_BitmapFile.GetAsAnimation(m_Bitmaps, m_FrameCount);
+		m_MainBitmap = m_Bitmaps.at(0);
+
+		if (m_FrameCount > 1) {
+			// If animation mode is set to something other than ALWAYSLOOP but only has 2 frames, override it because it's pointless
+			if (m_FrameCount == 2 && m_SpriteAnimMode >= SpriteAnimMode::ALWAYSRANDOM) { m_SpriteAnimMode = SpriteAnimMode::ALWAYSLOOP; }
+		} else {
+			m_SpriteAnimMode = SpriteAnimMode::NOANIM;
+		}
+		// Sampled color at the edges of the layer that can be used to fill gap if the layer isn't large enough to cover a target bitmap.
+		if (!m_WrapX) {
+			m_FillColorLeft = _getpixel(m_MainBitmap, 0, m_MainBitmap->h / 2);
+			m_FillColorRight = _getpixel(m_MainBitmap, m_MainBitmap->w - 1, m_MainBitmap->h / 2);
+		}
+		if (!m_WrapY) {
+			m_FillColorUp = _getpixel(m_MainBitmap, m_MainBitmap->w / 2, 0);
+			m_FillColorDown = _getpixel(m_MainBitmap, m_MainBitmap->w / 2, m_MainBitmap->h - 1);
+		}
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int SLBackground::Create(const SLBackground &reference) {
+		SceneLayer::Create(reference);
+
+		// The main bitmap is created and owned by SceneLayer because it can be modified. We need to destroy it to avoid a leak because the bitmaps we'll be using here are owned by ContentFile static maps and are unmodifiable.
```

Comment:
Pretty sure I can't. The flag is being set by `SeneLayer::Create(&reference)` so setting it before will be overridden.
And it's being set true there because iirc the reference already has a `m_MainBitmap` when we're doing the clone create, so a copy of the bitmap is being created.

I agree it's a bit of an ass setup, but un-assing it basically means having a 95% dupe of `SeneLayer` as a new base, or just copypasta 95% of it into this.

Reply:
I'll be honest, I barely remember this stuff now, 2.5 months later. So I'll take your word it's ass to do, feel free to resolve lol